fabric-recipe-api-v1/src/main/java/net/fabricmc/fabric/impl/recipe/sync/RecipeSyncRequestPayloadC2S.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.recipe.sync;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import net.minecraft.network.PacketByteBuf;
+import net.minecraft.network.codec.PacketCodec;
+import net.minecraft.network.codec.PacketCodecs;
+import net.minecraft.network.packet.CustomPayload;
+import net.minecraft.util.Identifier;
+
+public record RecipeSyncRequestPayloadC2S(Set<Identifier> synchronizedSerializers) implements CustomPayload {
```

Comment:
Nit: A comment documenting what these payloads are for might be useful, took me a little bit to figure out why we need a C2S packet.